src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -79,16 +107,67 @@ internal static class IntellisenseQuickInfoBuilder
             }
 
             var content = new ContainerElement(
-                                ContainerElementStyle.Stacked,
+                                ContainerElementStyle.Stacked | ContainerElementStyle.VerticalPadding,
                                 elements);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
 
-        private static ClassifiedTextElement BuildClassifiedTextElement(QuickInfoSection section)
+        private static IEnumerable<object> BuildClassifiedTextElements(QuickInfoSection section)
         {
-            return new ClassifiedTextElement(section.TaggedParts.Select(
-                    part => new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text)));
+            var paragraphs = new List<object>();
+            var currentParagraph = new List<object>();
+            var currentRuns = new List<ClassifiedTextRun>();
+            foreach (var part in section.TaggedParts)
```

Comment:
it would be nice if you can explain what this is actually trying to do.

Reply:
and this code seems require some testing on various different input cases.

Reply:
Lots of comments coming in the next push.

Reply:
Now documented. Marking this comment resolved; we have a couple other open comments tracking my work to add more tests.